File: postgres_39/all_struct_with_default/ResentVolSummary.go
package dto

type Resent_vol_summary struct {
	Resent_id                string   `json:"resent_id" default:"source"`
	Reserve_class_id         string   `json:"reserve_class_id" default:"source"`
	Product_type             string   `json:"product_type" default:"source"`
	Summary_id               string   `json:"summary_id" default:"source"`
	Summary_obs_no           int      `json:"summary_obs_no" default:"1"`
	Active_ind               *string  `json:"active_ind" default:""`
	Analyst_ba_id            *string  `json:"analyst_ba_id" default:""`
	Approved_by_ba_id        *string  `json:"approved_by_ba_id" default:""`
	Approved_date            *string  `json:"approved_date" default:""`
	Created_date             *string  `json:"created_date" default:""`
	Created_date_desc        *string  `json:"created_date_desc" default:""`
	Current_balance          *float64 `json:"current_balance" default:""`
	Current_balance_ouom     *string  `json:"current_balance_ouom" default:""`
	Current_balance_uom      *string  `json:"current_balance_uom" default:""`
	Decline_case_id          *string  `json:"decline_case_id" default:""`
	Effective_date           *string  `json:"effective_date" default:""`
	Expiry_date              *string  `json:"expiry_date" default:""`
	Gross_summary_ind        *string  `json:"gross_summary_ind" default:""`
	Interest_set_id          *string  `json:"interest_set_id" default:""`
	Interest_set_partner     *string  `json:"interest_set_partner" default:""`
	Interest_set_seq_no      *int     `json:"interest_set_seq_no" default:""`
	Material_balance_case_id *string  `json:"material_balance_case_id" default:""`
	Net_summary_ind          *string  `json:"net_summary_ind" default:""`
	Open_balance             *float64 `json:"open_balance" default:""`
	Open_balance_ouom        *string  `json:"open_balance_ouom" default:""`
	Open_balance_uom         *string  `json:"open_balance_uom" default:""`
	Partner_obs_no           *int     `json:"partner_obs_no" default:""`
	Pden_id                  *string  `json:"pden_id" default:""`
	Pden_product_type        *string  `json:"pden_product_type" default:""`
	Pden_source              *string  `json:"pden_source" default:""`
	Pden_subtype             *string  `json:"pden_subtype" default:""`
	Ppdm_guid                *string  `json:"ppdm_guid" default:""`
	Remark                   *string  `json:"remark" default:""`
	Report_ind               *string  `json:"report_ind" default:""`
	Source                   *string  `json:"source" default:""`
	Volume_method            *string  `json:"volume_method" default:""`
	Vol_anal_case_id         *string  `json:"vol_anal_case_id" default:""`
	Yield_parent_product     *string  `json:"yield_parent_product" default:""`
	Yield_rate               *float64 `json:"yield_rate" default:""`
	Yield_rate_ouom          *string  `json:"yield_rate_ouom" default:""`
	Yield_rate_uom           *string  `json:"yield_rate_uom" default:""`
	Row_changed_by           *string  `json:"row_changed_by" default:""`
	Row_changed_date         *string  `json:"row_changed_date" default:""`
	Row_created_by           *string  `json:"row_created_by" default:""`
	Row_created_date         *string  `json:"row_created_date" default:""`
	Row_effective_date       *string  `json:"row_effective_date" default:""`
	Row_expiry_date          *string  `json:"row_expiry_date" default:""`
	Row_quality              *string  `json:"row_quality" default:""`
}

File: create_tables/resent_vol_summary.py
import oracledb
import sys
import lib_platform as platform

if platform.system == "darwin":
    sys.path.insert(0, "database")
elif platform.system == "windows":
    sys.path.insert(0, "../database")

from conn import connection

# pw = getpass.getpass("Enter password: ")

# connection = oracledb.connect(
#     user="admin",
#     password="ppdm-3.9",
#     dsn="/kei-oracle-test.chkxr21l2l5s.ap-southeast-3.rds.amazonaws.com:1521/ORCL")

# print("Successfully connected to Oracle Database")

cursor = connection.cursor()


cursor.execute("""
    begin
        execute immediate 'drop table resent_vol_summary';
        exception when others then if sqlcode <> -942 then raise; end if;
    end;""")

cursor.execute("""
    create table resent_vol_summary (
        RESENT_ID VARCHAR(40) PRIMARY KEY NOT NULL,
        RESERVE_CLASS_ID VARCHAR(40) NOT NULL,
        PRODUCT_TYPE VARCHAR(40) NOT NULL,
        SUMMARY_ID VARCHAR(40) NOT NULL,
        SUMMARY_OBS_NO INT,
        ACTIVE_IND VARCHAR(1),
        ANALYST_BA_ID VARCHAR(40),
        APPROVED_BY_BA_ID VARCHAR(40),
        APPROVED_DATE TIMESTAMP(0),
        CREATED_DATE TIMESTAMP(0),
        CREATED_DATE_DESC VARCHAR(8),
        CURRENT_BALANCE DECIMAL(14,4),
        CURRENT_BALANCE_OUOM VARCHAR(40),
        CURRENT_BALANCE_UOM VARCHAR(40),
        DECLINE_CASE_ID VARCHAR(40),
        EFFECTIVE_DATE TIMESTAMP(0),
        EXPIRY_DATE TIMESTAMP(0),
        GROSS_SUMMARY_IND VARCHAR(1),
        INTEREST_SET_ID VARCHAR(40),
        INTEREST_SET_PARTNER VARCHAR(40),
        INTEREST_SET_SEQ_NO INT,
        MATERIAL_BALANCE_CASE_ID VARCHAR(40),
        NET_SUMMARY_IND VARCHAR(1),
        OPEN_BALANCE DECIMAL(14,4),
        OPEN_BALANCE_OUOM VARCHAR(40),
        OPEN_BALANCE_UOM VARCHAR(40),
        PARTNER_OBS_NO INT,
        PDEN_ID VARCHAR(40),
        PDEN_PRODUCT_TYPE VARCHAR(40),
        PDEN_SOURCE VARCHAR(40),
        PDEN_SUBTYPE VARCHAR(30),
        PPDM_GUID VARCHAR(38),
        REMARK VARCHAR(2000),
        REPORT_IND VARCHAR(1),
        SOURCE VARCHAR(40),
        VOLUME_METHOD VARCHAR(40),
        VOL_ANAL_CASE_ID VARCHAR(40),
        YIELD_PARENT_PRODUCT VARCHAR(40),
        YIELD_RATE DECIMAL(14,4),
        YIELD_RATE_OUOM VARCHAR(40),
        YIELD_RATE_UOM VARCHAR(40),
        ROW_CHANGED_BY VARCHAR(30),
        ROW_CHANGED_DATE TIMESTAMP(0),
        ROW_CREATED_BY VARCHAR(30),
        ROW_CREATED_DATE TIMESTAMP(0),
        ROW_EFFECTIVE_DATE TIMESTAMP(0),
        ROW_EXPIRY_DATE TIMESTAMP(0),
        ROW_QUALITY VARCHAR(40))""")
